### PR TITLE
Mn 157 - doc public methods Attestation

### LINKED ISSUE #157: DOC - API Doc for public methods on Attestation
Should include examples 

See https://github.com/KILTprotocol/sdk-js/pull/174

### PR DESCRIPTION
**Pull request by maudnals**
## How to test:  
If you'd like to see how the doc looks like:  
Run `npm run build:docs`. This generates the doc at `sdk-js/docs/api`.
Open the so-generated files in your browser: `sdk-js/docs/api/classes/attestation.attestation-1.html`.

## Checklist:

- [x] I have verified that the code works // tests ran
- [x] I have verified that the code is easy to understand
  - [x] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

File: src/attestation/Attestation.ts [hunk 1]
```diff
@@ -58,25 +53,143 @@ export default class Attestation implements IAttestation {
...
+   * ```
+   */
+  public static async query(claimHash: string): Promise<Attestation | null> {
+    return query(claimHash)
+  }
+
+  /**
+   * @description (STATIC) (ASYNC) Revokes an attestation.
+   * @param claimHash The hash of the claim that corresponds to the attestation to revoke.
+   * @param identity The identity used to revoke the attestation (should be an attester identity).
```
Comment (Dudleyneedham) [inline src/attestation/Attestation.ts]: (should be an identity that has been delegated rights).

Comment (maudnals) [reply src/attestation/Attestation.ts]: Great catch, thanks 👍   
Updated

File: src/attestation/Attestation.ts [hunk 2]
```diff
@@ -58,25 +53,142 @@ export default class Attestation implements IAttestation {
     this.revoked = revoked
   }
 
+  /**
+   * @description (STATIC) (ASYNC) Queries the chain about a given attestation, by `claimHash`.
+   * @param claimHash The hash of the claim that corresponds to the attestation to query.
+   * @returns A promise containing the [[Attestation]] or `null`.
+   * @example
+   * ```javascript
+   * const attestation = await Attestation.query("0xd810224b1b6a4db8d1d1e909d1aeb7d441846914ed024cdc147c4fa9221cd177");
```
[the remark this answers is not in the document]
Comment (maudnals) [reply src/attestation/Attestation.ts]: ok, addressed in weekly

File: src/attestation/Attestation.ts [hunk 3]
```diff
@@ -58,25 +53,142 @@ export default class Attestation implements IAttestation {
...
+    // Query attestation by claimHash. null if no attestation is found on-chain for this hash
     const attestation: Attestation | null = await query(claimHash)
-    // 2) check attestation for being valied, having the correct owner and not being revoked
-    const attestationValid: boolean =
+    // Check if attestation is valid
+    const isValid: boolean = this.isAttestationValid(attestation)
+    if (!isValid) {
+      log.debug(() => 'No valid attestation found')
+    }
+    return Promise.resolve(isValid)
```
[the remark this answers is not in the document]
Comment (maudnals) [reply src/attestation/Attestation.ts]: OK answered live by @tjwelde: there's no reason, we should fix this.

File: src/attestation/Attestation.ts [hunk 4]
```diff
@@ -1,15 +1,17 @@
 /**
- * Attestations are used to certify [[Claim]] objects, which are then written on the [[Blockchain]].
- * ***
- *  Attestation issued by the Attester is sent to and stored with the claimer. We call these [[AttestedClaim]]s "Credentials".
+ * An [[Attestation]] certifies a [[Claim]], sent by a claimer in the form of a [[RequestForAttestation]]. [[Attestation]]s are **written on the blockchain** and are **revokable**.
+ * Note: once an [[Attestation]] is stored, it can be sent to and stored with the claimer as an [[AttestedClaim]] (= "Credential").
```
Comment (Dudleyneedham) [inline src/attestation/Attestation.ts]: Is the equals meant to be there?

Comment (maudnals) [reply src/attestation/Attestation.ts]: fixed, ty

**Pull request merged**